integration_tests/run_pyspark_from_build.sh

```diff
@@ -333,6 +333,11 @@ EOF
                 --driver-class-path "${PYSP_TEST_spark_driver_extraClassPath}"
                 --conf spark.executor.extraClassPath="${PYSP_TEST_spark_driver_extraClassPath}"
             )
+        elif [[ -n "$PYSP_TEST_spark_jars_packages" ]]; then
+            SPARK_SHELL_ARGS_ARR+=(--packages "${PYSP_TEST_spark_jars_packages}")
+            if [[ -n "$PYSP_TEST_spark_jars_repositories" ]]; then
+                SPARK_SHELL_ARGS_ARR+=(--repositories "${PYSP_TEST_spark_jars_repositories}")
+            fi
```

Comment:
This should not be nested. If specified, we should convey to the cmdline even if packages not specified.  Also matches the pattern used in the non-smoke section below this (would be nice to not replicate this code).
```suggestion
          fi
          if [[ -n "$PYSP_TEST_spark_jars_repositories" ]]; then
              SPARK_SHELL_ARGS_ARR+=(--repositories "${PYSP_TEST_spark_jars_repositories}")
          fi
```